### PR TITLE
minor htaccess changes

### PR DESCRIPTION
**Pull request by mightyBroccoli**
I added some htaccess files to restrict the access to files the web server should not present to the outside (log files, script files e.g. ).
Furthermore I edited the gitignore files for avatars and logs to even allow to htaccess to function properly. Removing the wildcard for ignoring anything and replacing it with a more specific line.

> The .htaccess file on root level is with release 1.1.0 unnecessary.

@Newcomer1989  You said in my last pull request that the htaccess in the root dir was not neccecary but now there is not even a index.php inside root to do anything so the webserver would just list the hosted directory. I dont know if I possibly did something wrong so that would be just temporarily. I just redirect from root directly to stats without any reloading or anything.

### CODE REVIEW

**Comment by Shad86** [comment]
i saw in the "webinterface" folder a "index.php" maybe this do the trick for you :)

**Comment by mightyBroccoli** [comment]
yes in both stats and webinterface folders is an index.php file and that is correct but the point is you need to configure your apache or what ever to the root directory. 
When you do that there is no index.php file to load anything you just get a listing of all files and folder in that webspace. Thats why I introduced the redirect through htaccess to automatically get the user to /stats/. 
The other problem is when you just point to /stats/ and not to the root directory your css files dont properly work and you get no style sheet, and the even bigger problem is you need to get another webhost to configure your bot because your cant access /webinterface/ from inside /stats/.

**Comment by Shad86** [comment]
Im not that into but could it may work with symlinks? so that you can access through symlink into that folder - is just a thought - as i said im not that into (sorry)

**Comment by mightyBroccoli** [comment]
Symlinks would be possible but symlinks are thier own territory. I really dont want to mess with them. When you configure them just a little but wrong, the possible gap in your security could be humongous. 
Redirection is easier and the time to load the target site is much lower.
I could implement some index.php file inside the root directory to make it a lot smoother but my php is really bad so I could just offer index.html

Something like 
```
<!DOCTYPE HTML>
<html lang="de-DE">
    <head>
        <meta charset="UTF-8">
        <meta http-equiv="refresh" content="1;url=http://magicbroccoli.de/stats/">
        <script type="text/javascript">
            window.location.href = "http://magicbroccoli.de/stats/"
        </script>
    </head>
    <body>
        If you are not redirected automatically, click the fucking <a href='http://magicbroccoli.de/stats/>link to it.</a>
    </body>
</html>
```

**Comment by Newcomer1989** [comment]
I do not agree with all of your pull request. The access to the file list on the root directory is a wrong configuration of your webserver. A redirect on this point is not needed and might will be confuse other user.
The next days I will have a look to this and perhaps a part of it I will add to the next release version. The thing what I have to overtake for is: Have I to decide for the Ranksystem user, what directory should be forbidden or is this the job of each webspace/webserver administrator himself?! Also not all webserver supports a .htaccess file. So there will come issues like, htaccess is not working (I am using nginx), and so on..

**Comment by mightyBroccoli** [comment]
@Newcomer1989 In my case it is an apache configuration. I really dont see how when my apache is pointing to the root dir, how the page should load. There are no files to start with. The browsing software does not have any idea what to do, thats why the file list is presented. 
The decision to forbid some directories is natural. Without any boundaries the webuser can just browser through your config files without any authentication. Same for the log directory. 
Your point about htaccess could not be working that is true when the hosting software is nginx which does not support htaccess, there the webadmin has to block access within the configuration of ngnix but the htaccess does not harm or hinder any other hosting software.

In conclusion there are just 2 sides the user is hosting with apache or anything else then ngnix then the htaccess would work perfectly. I designed them to accept versions from really old to fairly new and up to date.
The other side would be ngnix there the admin has to configure him/herself what he/she wants to show to ppl. 

BUT with the webspace poiting to the root dir there will be no page loading without the redirect because there is nothing to load there.
